### PR TITLE
Add NoisePerturbAugmentor and CHiME3 data preparation.

### PR DESCRIPTION
**Pull request by xinghai-sun**
1. Add NoisePerturbAugmentor and ImpulseResponseAugmentor.
2. CHiME3 data preparation.

### CODE REVIEW

File: deep_speech_2/train.py [hunk 1]
```diff
@@ -110,9 +110,7 @@
     "the existing model of this path. (default: %(default)s)")
 parser.add_argument(
     "--augmentation_config",
-    default='[{"type": "shift", '
-    '"params": {"min_shift_ms": -5, "max_shift_ms": 5},'
-    '"prob": 1.0}]',
+    default=open('augmentation.config', 'r').read(),
```
Comment (pkuyym) [inline deep_speech_2/train.py]: better to mv ```augmentation.config``` to a unified directory like ```conf```

File: deep_speech_2/augmentation.config [hunk 2]
```diff
@@ -0,0 +1,34 @@
+[
```
Comment (pkuyym) [inline deep_speech_2/augmentation.config]: We can create a directory to hold configuration files.

File: deep_speech_2/data_utils/augmentor/impulse_response.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+class ImpulseResponseAugmentor(AugmentorBase):
+    """Augmentation model for adding impulse response effect.
+    
+    :param rng: Random generator object.
+    :type rng: random.Random
+    :param impulse_manifest: Manifest path for impulse audio data.
+    :type impulse_manifest: basestring 
+    """
+
+    def __init__(self, rng, impulse_manifest):
```
Comment (pkuyym) [inline deep_speech_2/data_utils/augmentor/impulse_response.py]: ```impulse_manifest_path``` is better

File: deep_speech_2/data_utils/augmentor/noise_perturb.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    :type rng: random.Random
+    :param min_snr_dB: Minimal signal noise ratio, in decibels.
+    :type min_snr_dB: float
+    :param max_snr_dB: Maximal signal noise ratio, in decibels.
+    :type max_snr_dB: float
+    :param noise_manifest: Manifest path for noise audio data.
+    :type noise_manifest: basestring 
+    """
+
+    def __init__(self, rng, min_snr_dB, max_snr_dB, noise_manifest):
```
Comment (pkuyym) [inline deep_speech_2/data_utils/augmentor/noise_perturb.py]: ```noise_manifest_path``` is better

File: deep_speech_2/data_utils/data.py [hunk 5]
```diff
@@ -202,6 +202,7 @@ def _process_utterance(self, filename, transcript):
         """Load, augment, featurize and normalize for speech data."""
         speech_segment = SpeechSegment.from_file(filename, transcript)
         self._augmentation_pipeline.transform_audio(speech_segment)
+        speech_segment.to_wav_file("audio.test.wav")
```
Comment (pkuyym) [inline deep_speech_2/data_utils/data.py]: Maybe we can save ```audio.test.wav```  to a temp directory

File: deep_speech_2/datasets/noise/chime3_background.py [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+            os.path.join(target_dir, 'CHiME3_background_caf.zip'), target_dir)
+        unpack(
+            os.path.join(target_dir, 'CHiME3_background_ped.zip'), target_dir)
+        unpack(
+            os.path.join(target_dir, 'CHiME3_background_str.zip'), target_dir)
+    else:
+        print("Skip downloading and unpacking. Data already exists in %s." %
+              target_dir)
+    # create manifest json file
+    create_manifest(target_dir, manifest_path)
```
Comment (pkuyym) [inline deep_speech_2/datasets/noise/chime3_background.py:116]: maybe we should do some checking here to make sure existence of audio files and transcription text files

**Review by pkuyym — COMMENTED** [review]
Almost LGTM.

File: deep_speech_2/augmentation.config (hunk 2 above)
Comment (xinghai-sun) [reply deep_speech_2/augmentation.config]: Moved to `conf/`.

File: deep_speech_2/data_utils/augmentor/noise_perturb.py (hunk 4 above)
Comment (xinghai-sun) [reply deep_speech_2/data_utils/augmentor/noise_perturb.py]: Done.

File: deep_speech_2/data_utils/augmentor/impulse_response.py (hunk 3 above)
Comment (xinghai-sun) [reply deep_speech_2/data_utils/augmentor/impulse_response.py]: Done.

File: deep_speech_2/data_utils/data.py (hunk 5 above)
Comment (xinghai-sun) [reply deep_speech_2/data_utils/data.py]: Removed.

File: deep_speech_2/datasets/noise/chime3_background.py (hunk 6 above)
Comment (xinghai-sun) [reply deep_speech_2/datasets/noise/chime3_background.py:116]: 不需要，第一，noise文件不存在transcription；第二，这里的逻辑是是遍历所有的音频文件（不存在就不会被遍历到）。

File: deep_speech_2/train.py (hunk 1 above)
Comment (xinghai-sun) [reply deep_speech_2/train.py]: Done.

**Review by pkuyym — APPROVED** [review]

**Pull request merged**